### DISCUSSION TITLE
[RFC] 069 - V2 Breaking Change

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#3283] [RFCs]
## 背景

这里记录一些关于 V1 中的历史债务，考虑在 V2 中做 Breaking Change 

## Breaking Change

- [x] basePath： 实现不完整，代码量又冗余。考虑去掉 https://github.com/lobehub/lobe-chat/pull/9015
- ~~swr -> react-query： swr 在做一些体验进阶的场景下有点力不从心（例如无限分页），接口设计没有 react-query 优雅。而 react-query 和 trpc 的兼容度更好，如果用使用 trpc 其实就不应该有 swr 什么事情了。~~  实际实现下来 react-query 并没有比 swr 好太多，这一条废弃
- [x] 移除 azure-ad 模块：微软方案变更，https://github.com/lobehub/lobe-chat/pull/4579 ，v2 移除 azure-ad 的 auth 模块, https://github.com/lobehub/lobe-chat/pull/9942
- [x] 移除基于 WebRTC 的同步逻辑实现： 云端版使用 db 更理想 ~~客户端版本替换为 Webdav #5261  https://github.com/lobehub/lobe-chat/pull/9002~~
- [x] ~~废弃 dexie ，本地 db 全量使用 pglite https://github.com/lobehub/lobe-chat/issues/5131~~ 不再支持 浏览器 client 模式 https://github.com/lobehub/lobe-chat/pull/9991
- [x] 移除所有 next-auth 相关的 authEnv 环境变量，全走 inference 方式，相关 PR： #3701  https://github.com/lobehub/lobe-chat/pull/9954
- [x] 将 `S3_SET_ACL` 默认值设为 1 ，S3 默认使用加签模式，这样 `S3_PUBLIC_DOMAIN` 就变成可选项。 
  - 感谢 @vual 的 #4254 的 PR
  - 相关讨论：#3753
- [x] 类型定义 + 校验 schema 做一套完整的方案（考虑单独出 types？)，确保 client 到 trpc 的请求均通过完善校验；https://github.com/lobehub/lobe-chat/pull/10016
- [x] 移除 v1 客户端 DB 时期的迁移代码 https://github.com/lobehub/lobe-chat/pull/9991
- [x] 移除 /settings/llm 页面 https://github.com/lobehub/lobe-chat/pull/9940
- [x] 约束最小浏览器版本： Chrome 99 ，Safari 16.4
- [x] 移除 Doubao Provider（已经重命名为火山引擎 https://github.com/lobehub/lobe-chat/pull/6221 ）
- [x] Feature Flag 变更
  - [x] 移除 `clerk-sign-up` feature flag，由于 clerk 已经支持 restrict 模式来关闭注册 https://github.com/lobehub/lobe-chat/pull/10001
  - [x] 重命名 `language_model_settings` 为 `provider_settings` https://github.com/lobehub/lobe-chat/pull/10264
- [x] 在 OpenAI 渠道将 Responses API 设为默认 api mode https://github.com/lobehub/lobe-chat/pull/10202
- [x] 移除纯 client 模式 https://github.com/lobehub/lobe-chat/pull/10001

## 依赖大版本升级

- [x] jose -> v6： 之前升级导致某些场景有问题，回滚了
- [x] vitest -> v3

**Comment by ous50** [comment]
> - [ ] 移除基于 WebRTC 的同步逻辑实现： 云端版使用 db 更理想；

我觉得这个对于开源版还是可以保留的，毕竟云端版对公众开放也挺麻烦（）

**Reply by arvinxx** [reply]
不会考虑了，这个功能哪怕不开启也会带来 50~100kb 左右的 gzip 体积增加和很大的服务端/客户端内存占用

> 毕竟云端版对公众开放也挺麻烦（）

开源版关注的是个人使用场景，你自己开放给公众使用的这种场景就不是开源版核心关注的点。

**Comment by SoulGolf** [comment]
多模型对比回答在V2的计划中吗

**Reply by arvinxx** [reply]
在。另外对比会不仅限模型的对比，理论上是各种配置的组合对比

**Reply by SoulGolf** [reply]
> 在。另外对比会不仅限模型的对比，理论上是各种配置的组合对比

有预期的pr时间点吗

**Reply by arvinxx** [reply]
看3月有没有机会搞出来吧。如果来不及 4月肯定可以

**Comment by sssession** [comment]
> 移除 /settings/llm 页面

为什么要移除这个页面？

**Reply by arvinxx** [reply]
> > 移除 /settings/llm 页面
> 
> 
> 
> 为什么要移除这个页面？

因为切到 /settings/provider了

**Comment by gngshn** [comment]
请问下现在是桌面端开发优先于 V2 版本吗?
谢谢

**Reply by arvinxx** [reply]
桌面端发布后，接下来就是 V2

**Reply by m0eak** [reply]
啥时候整多模态绘图啊

**Comment by htdohk** [comment]
虽然我一直很坚定的使用Lobe-chat、并且向朋友同事都在推荐，不过在目前越来越多厂商的会员订阅支持代理模式了，这对开源前端产生了一些冲击。照此发展Lobe-chat的使用场景会有些局限。比如Artifacts一直是个很不错的自带插件，拓宽了交付结果的形式。所以可否考虑再引入一些类似内置插件，可以实现Shell、Python、浏览器、甚至虚拟机的功能呢。不然总觉得这个2.0是1.0的优化、精简、重构，而不是feature 2.0。。。

**Reply by arvinxx** [reply]
这个是技术层面的变更。feature 上今天会发一个 Announcement 出来

**Reply by arvinxx** [reply]
@htdohk 详见： https://github.com/lobehub/lobe-chat/discussions/10007

**Comment by CanYuanA** [comment]
> * 将 `S3_SET_ACL` 默认值设为 1 ，S3 默认使用加签模式，这样 `S3_PUBLIC_DOMAIN` 就变成可选项。

对于这部分我有一些疑问，根据当前实现，似乎S3_SET_ACL=0时才会使用S3的加签模式；且就算使用加签模式应该还是需要S3_PUBLIC_DOMAIN这个值的（作为访问链接前缀）。我看这一条已经打勾了，目前实现是否和预期不符？
